### ISSUE TITLE
Suggestion: Add progress/activity bar while scanning

### ISSUE THREAD

**Issue by bubbob42** [Arisamiga/Mnemosyne#9]
Scanning dirs is fast on "high end" classic machines, but could potentially take a long time on slower ones.

Suggestion: Add an activity animation or (progress) bar during scan.

**Comment by Arisamiga** [comment]
Thank you for your suggestion!

Unfortunately, this is not possible with how Mnemosyne works as it scans the directories recursively so it does not know what the total of the directory is either in size or files.

As this is not possible I am closing this Issue if you have any further comments or issues don't hesitate to re-open it!

**Comment by bubbob42** [comment]
Hi!

It is not necessary to show progress. But is would be nice if users knew that the tool didn't freeze. Currently, one has to check HD activity LED for this (which might not be connected on some HD controllers).

IMO it would be sufficient to
- either show the current scan path in the window title while scanning or open a subwindow displaying the current scan path or
- even simpler: just display "scanning..." somewhere (you could animate the "...")

**Comment by Arisamiga** [comment]
> Hi!
> 
> It is not necessary to show progress. But is would be nice if users knew that the tool didn't freeze. Currently, one has to check HD activity LED for this (which might not be connected on some HD controllers).
> 
> IMO it would be sufficient to
> 
> * either show the current scan path in the window title while scanning or open a subwindow displaying the current scan path or
> * even simpler: just display "scanning..." somewhere (you could animate the "...")

Hello! Mnemosyne already has one of the features which basically displays "Scanning" followed by the name of the directory you are scanning.

![mnemosyne_scanning](https://github.com/Arisamiga/Mnemosyne/assets/64918822/1b229fef-7aa0-4df6-8f5b-7203f23d9609)

**Comment by bubbob42** [comment]
> > Hi!
> > It is not necessary to show progress. But is would be nice if users knew that the tool didn't freeze. Currently, one has to check HD activity LED for this (which might not be connected on some HD controllers).
> > IMO it would be sufficient to
> > 
> > * either show the current scan path in the window title while scanning or open a subwindow displaying the current scan path or
> > * even simpler: just display "scanning..." somewhere (you could animate the "...")
> 
> Hello! Mnemosyne already has one of the features which basically displays "Scanning" followed by the name of the directory you are scanning.

Yes, but at the moment this will display the root path. If we scan a large partition (could take half an hour or more), it'll remain static.

**Comment by Arisamiga** [comment]
closed by https://github.com/Arisamiga/Mnemosyne/pull/20
